mkosi/config.py

```diff
@@ -456,7 +456,7 @@ def config_parse_filename(value: Optional[str], old: Optional[str]) -> Optional[
         die(". and .. are not valid filenames")
 
     if "/" in value:
-        die(f"{value} is not a valid filename")
+        die(f"{value!r} is not a valid filename. (Output= requires a filename with no path components, relative to output directory.)")
```

Comment:
We should probably start passing in the setting name (either CLI switch or config file setting) to the parser functions. Then again though, we'd likely also want the file we're parsing and the line so maybe it's better to start throwing more recognizable exceptions which we catch in parse_config() and then show a proper error message that includes more information.